File: src/main/java/hu/crs/codewars/sevenkyu/jadencasestrings/Kata.java
package hu.crs.codewars.sevenkyu.jadencasestrings;

/**
 * Jaden case strings
 *
 * https://www.codewars.com/kata/5390bac347d09b7da40006f6
 */

public class Kata {
    public String toJadenCase(String phrase) {
        if (phrase == null || phrase.isEmpty()) {
            return null;
        }

        String[] phraseArray = phrase.split(" ");
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < phraseArray.length; i++) {
            sb.append(capitalize(phraseArray[i]));
            sb.append(" ");
        }

        return sb.subSequence(0, sb.length() - 1).toString();
    }

    private static String capitalize(String s) {
        return Character.toUpperCase(s.charAt(0)) + s.substring(1);
    }
}

File: src/main/kotlin/hu/crs/codewars/sevenkyu/shortestword/Kata.kt
package hu.crs.codewars.sevenkyu.shortestword

/**
 * Shortest Word
 *
 * https://www.codewars.com/kata/57cebe1dc6fdc20c57000ac9
 */
fun findShort(s: String): Int {
    return s.split(" ")
            .map { word -> word.length }
            .min()!!
}